### ISSUE TITLE
Install fails with nginx

### ISSUE THREAD

**Issue by Dreamsorcerer** [matomo-org/matomo-package#68]
Trying to install the package gives me this error:

```
Setting up matomo (3.5.1-2) ...
/var/lib/dpkg/info/matomo.postinst: 70: /var/lib/dpkg/info/matomo.postinst: a2enconf: not found
dpkg: error processing package matomo (--configure):
 subprocess installed post-installation script returned error exit status 127
Processing triggers for php7.0-fpm (7.0.30-0+deb9u1) ...
Processing triggers for libc-bin (2.24-11+deb9u3) ...
Setting up php-curl (1:7.0+49) ...
Errors were encountered while processing:
 matomo
E: Sub-process /usr/bin/dpkg returned an error code (1)
```

Obviously the problem is caused by me having nginx installed, not Apache.

On a side note, the Github URL is wrong on this page: http://debian.piwik.org/

**Comment by Dreamsorcerer** [comment]
Actually, the URLs are wrong all over the place, the README links are wrong here as well: https://github.com/matomo-org/matomo-package

**Comment by Dreamsorcerer** [comment]
OK, taking a little look into your postinst script, I see you are testing if the apache config dir exists. So, it looks like this is clashing with the php-fpm packgage, which installs /etc/apache2/conf-available/php7.0-fpm.conf regardless of whether Apache is installed or not.

Manually deleting the directory before installing Matomo provides a workaround.

**Comment by aureq** [comment]
@Dreamsorcerer at this point, nginx is not support. Also having more information about your environment would help. Finally, I'd be interested to see your nginx configuration.

**Comment by Dreamsorcerer** [comment]
For this bug, the issue is not setting up nginx, I'm not expecting you to support nginx (and I'd rather customise the server config myself anyway).

But, I expect the package not to error and fail to install just because I don't have Apache installed. As I mentioned in the previous comment, this seems to be a clash with the php-fpm package which installs an Apache config file regardless of whether Apache is installed or not.

Installing this package then checks if the Apache config directory exists (which it does thanks to php-fpm), and install another config. The error happens when it tries to run a2enconf, which does not exist, because apache2 is not installed.

I'm sure there are a number of possible solutions here, such as checking if a2enconf is available, before running it, or being more conservative and not running a2enconf at all, and just let the user enable the config once they're happy with it, etc.

**Comment by Findus23** [comment]
In case you are interested:
I have completely rewritten the official Matomo nginx config:
https://github.com/matomo-org/matomo-nginx/pull/43

If you have feedback, just comment there.

**Comment by Dreamsorcerer** [comment]
But, as per https://github.com/matomo-org/matomo-package/issues/69 , I don't see how that config can work. If I set the root to /var/www/mysite/ I get 'File not found' (because that directory is empty). While setting it to /usr/share/matomo/ gives me errors about Matomo not being able to write to /usr/share/matomo/. I'm still completely clueless as to how anybody is supposed to configure a server to run Matomo from this Debian package.

**Comment by mackuba** [comment]
I had the same problem, I've added a check if `a2enconf` exists in the post-install script before it tries to configure Apache: https://github.com/matomo-org/matomo-package/pull/84

**Comment by mackuba** [comment]
A temporary workaround is to create an empty script at `/usr/bin/a2enconf` that just includes `#!/bin/bash` and nothing else.

**Comment by aureq** [comment]
This will be part of the 3.6.0 onward.
merged #84
